python/ray/_private/services.py

```diff
@@ -1993,20 +1994,29 @@ def determine_plasma_store_config(
                 )
             else:
                 plasma_directory = ray._private.utils.get_user_temp_dir()
-                logger.warning(
-                    "WARNING: The object store is using {} instead of "
-                    "/dev/shm because /dev/shm has only {} bytes available. "
-                    "This will harm performance! You may be able to free up "
-                    "space by deleting files in /dev/shm. If you are inside a "
-                    "Docker container, you can increase /dev/shm size by "
-                    "passing '--shm-size={:.2f}gb' to 'docker run' (or add it "
-                    "to the run_options list in a Ray cluster config). Make "
-                    "sure to set this to more than 30% of available RAM.".format(
-                        ray._private.utils.get_user_temp_dir(),
-                        shm_avail,
-                        object_store_memory * (1.1) / (2**30),
-                    )
+                recommend_size_gb = object_store_memory * 1.1 / (2**30)
+                warning_message = (
+                    f"This will harm performance! You may be able to free up "
+                    f"space by deleting files in {dev_shm}. If you are inside a "
+                    f"Docker container, you can increase {dev_shm} size by "
+                    f"passing '--shm-size={recommend_size_gb:.2f}gb' to "
+                    f"'docker run' (or add it to the run_options list in "
+                    f"a Ray cluster config). Make sure to set this to more "
+                    f"than 30% of available RAM."
                 )
+
+                if plasma_directory != dev_shm:
+                    logger.warning(
+                        f"WARNING: The object store is using {plasma_directory} "
+                        f"instead of {dev_shm} because {dev_shm} has only "
+                        f"{shm_avail} bytes available. " + warning_message
+                    )
+                else:
+                    logger.warning(
+                        f"WARNING: The object store is using {plasma_directory} "
```

Comment:
```suggestion
                        f"WARNING: The object store is using {dev_shm} "
```
nit: we know they're equal in this branch and this makes the code read more similarly to the resulting error message


I think this case should maybe just be an exception if it ever did happen though...

Reply:
why? the temp directory can be specified by user with env var. do we want to ban using `/dev/shm` as the temp dir?

Reply:
because in this case the `shm_avail` < `object_store_memory`. I think this will cause the subsequent `mmap` to allocate shared memory for the object store to fail

Reply:
yeah I never saw anyone using /dev/shm for temp dir. +1 raising an exception instead